django/core/management/__init__.py

```diff
@@ -312,8 +312,11 @@ def execute(self):
                     autoreload.check_errors(django.setup)()
                 except Exception:
                     # The exception will be raised later in the child process
-                    # started by the autoreloader.
-                    pass
+                    # started by the autoreloader. Pretend it didn't happen by
+                    # loading an empty list of applications.
+                    apps.all_models = defaultdict(OrderedDict)
+                    apps.app_configs = OrderedDict()
+                    apps.apps_ready = apps.models_ready = apps.ready = True
```

Comment:
This is a very awful hack. It leaves the models in an inconsistent state. The ORM is non-functional. The Python process just stays alive long enough to start the autoreloader, which is a separate module that doesn't use anything that depends on Django settings.

I'm not convinced it's a good idea to provide an API for that. I'd have to call it `pretend_app_loading_worked_even_though_it_definitely_did_not_and_god_only_knows_what_the_current_state_of_the_orm_is` ;-)